### PR TITLE
[CALCITE-4297] Allow BigQuery to parse and validate Niladic functions (Mr. Swett)

### PR DESCRIPTION
**Pull request by jswett77**
Before this change BigQuery was unable to validate niladic functions. After this change BigQuery can parse niladic functions, e.g. CURRENT_TIMESTAMP()

Also, BIgQuery supports the CURRENT_DATETIME function, which is timestamp without timezone. This change also adds CURRENT_DATETIME to SqlStdOperatorTable.

### CODE REVIEW

File: site/_docs/reference.md [hunk 1]
```diff
@@ -1291,6 +1291,7 @@ Not implemented:
 | LOCALTIMESTAMP(precision) | Returns the current date and time in the session time zone in a value of datatype TIMESTAMP, with *precision* digits of precision
 | CURRENT_TIME              | Returns the current time in the session time zone, in a value of datatype TIMESTAMP WITH TIME ZONE
 | CURRENT_DATE              | Returns the current date in the session time zone, in a value of datatype DATE
+| CURRENT_DATETIME          | Returns the current datetime in the session time zone, in a value of datatype TIMESTAMP
```
Comment (jswett77) [inline site/_docs/reference.md:1294]: Not sure if this makes sense, also when I added CURRENT_DATETIME to reserved words I broke `DocumentationTest.testGenerateKeyWords`

File: core/src/test/java/org/apache/calcite/tools/PlannerTest.java [hunk 2]
```diff
@@ -147,6 +148,21 @@ private void checkParseAndConvert(String query,
...
+        SqlParser.config());
+    SqlNode node = planner.parse("select current_datetime");
+    planner.validate(node);
+  }
+
+  @Test void testNiladicForBigQuery() throws Exception {
+    Planner planner = getPlanner(null,
+        SqlParser.config().withConformance(SqlConformanceEnum.BIG_QUERY));
+    SqlNode node = planner.parse("select current_time, current_time(), current_date, "
+            + "current_date(), current_timestamp, current_timestamp(), current_datetime, current_datetime()");
```
Comment (danny0405) [inline core/src/test/java/org/apache/calcite/tools/PlannerTest.java:162]: Maybe we should move the test to `SqlValidatorTest`

File: core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java [hunk 3]
```diff
@@ -1888,6 +1888,12 @@ public SqlSyntax getSyntax() {
   public static final SqlFunction CURRENT_TIMESTAMP =
       new SqlAbstractTimeFunction("CURRENT_TIMESTAMP", SqlTypeName.TIMESTAMP);
 
+  /**
+   * The <code>CURRENT_DATETIME [(<i>precision</i>)]</code> function.
+   */
+  public static final SqlFunction CURRENT_DATETIME =
+      new SqlAbstractTimeFunction("CURRENT_DATETIME", SqlTypeName.TIMESTAMP);
+
```
Comment (danny0405) [inline core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java:1896]: Does `CURRENT_DATETIME` belong to the SQL standard ?

Comment (jswett77) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java:1896]: No, I realized there is a better way to do this. I'll remove this change.

File: core/src/test/java/org/apache/calcite/tools/PlannerTest.java (hunk 2 above)
Comment (jswett77) [reply core/src/test/java/org/apache/calcite/tools/PlannerTest.java:162]: Ah, sure I can move it.

File: core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java [hunk 4]
```diff
@@ -360,6 +365,18 @@ private static String cannotStreamResultsForNonStreamingInputs(String inputs) {
...
 
+  @Test void testNiladicForBigQuery() throws Exception {
+    final SchemaPlus rootSchema = Frameworks.createRootSchema(true);
+    final FrameworkConfig config = Frameworks.newConfigBuilder()
+        .parserConfig(SqlParser.config().withConformance(SqlConformanceEnum.BIG_QUERY))
+        .defaultSchema(rootSchema)
+        .build();
+    Planner planner = Frameworks.getPlanner(config);
+    SqlNode node = planner.parse("select current_time, current_time(), current_date, "
+        + "current_date(), current_timestamp, current_timestamp()");
```
Comment (danny0405) [inline core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java:376]: You can just use `sql(xxx).withConformance(SqlConformanceEnum.BIG_QUERY).ok()`

**Review by danny0405 — APPROVED** [review]
+1, LGTM ~

**Pull request closed without merging**